### PR TITLE
WIP: Add docker bindings for easy management

### PR DESCRIPTION
**Pull request by Devp00l**
Do everything you need with one command:

* Spin up a cluster / multicluster
* Reset your cluster / multicluster
* Stop your cluster / multicluster
* Recompile ceph
* Start, stop, reset, test or debug ceph-dashboard

The zshrc is basically [grml's zshrc](https://raw.githubusercontent.com/grml/grml-etc-core/master/etc/zsh/zshrc) only at the end where the bindings are sourced is different.

It's very helpful surely not perfect but a start.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,14 +48,15 @@ Now start up the container, by mounting the local git clone directory as
 `/ceph`:
 
     # docker run -itd \
-      -v $PWD:/ceph \
-      -v ~/.ccache:/root/.ccache \
+      -v <pathToCephRepo>:/ceph \
+      -v $PWD/scripts:/scripts \
+      -v <pathToCCacheDir>:/root/.ccache \
```
Comment (ricardoasmarques) [inline README.md]: If you are replacing `~/.ccache`, you should adapt the documentation (see line 63)

File: README.md [hunk 2]
```diff
@@ -48,14 +48,15 @@ Now start up the container, by mounting the local git clone directory as
...
       --net=host \
       --name=ceph-dev \
       --hostname=ceph-dev \
       --add-host=ceph-dev:127.0.0.1 \
       ceph-dev-docker \
-      /bin/bash
+      /bin/zsh
```
Comment (ricardoasmarques) [inline README.md]: I'm getting the following error when trying to execute `docker run ...`:

````
docker: Error response from daemon: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/bin/zsh\": stat /bin/zsh: no such file or directory".
````

File: README.md [hunk 3]
```diff
@@ -48,8 +48,9 @@ Now start up the container, by mounting the local git clone directory as
 `/ceph`:
 
     # docker run -itd \
-      -v $PWD:/ceph \
-      -v ~/.ccache:/root/.ccache \
+      -v <pathToCephRepo>:/ceph \
+      -v $PWD/scripts:/scripts \
```
[the remark this answers is not in the document]
Comment (Devp00l) [reply README.md]: It's [not possible](https://docs.docker.com/engine/reference/builder/#notes-about-specifying-volumes) to specify a host directory inside the docker file.

I will replace the $PWD.

File: scripts/bashrc [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  cd $SDP
+}
+export -f cdsd
+
+function cdsdf {
+  cd $SDP/frontend
+}
+export -f cdsdf
+
+function ceph {
```
[the remark this answers is not in the document]
Comment (Devp00l) [reply scripts/bashrc]: All executables inside `/ceph/build/bin` will only use the vstart cluster if you are in the `/ceph/build` directory. If you for example call `ceph -s` anywhere else you will get errors.

File: scripts/bashrc [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  cdb
+  bin/ceph $@
+  cd -
+}
+export -f ceph
```
[the remark this answers is not in the document]
Comment (Devp00l) [reply scripts/bashrc]: I locally removed all exports again as you can only use `export -f` in bash, not in any other shells.

**Comment by Devp00l** [comment]
I will reopen this PR in a different version with less changes to get it in more smoothly

**Pull request closed without merging**